pallets/did/src/did_details.rs

```diff
@@ -634,32 +635,57 @@ pub struct DidUpdateDetails<T: Config> {
 	/// A new set of key agreement keys to add to the ones already stored.
 	pub new_key_agreement_keys: BTreeSet<DidEncryptionKey>,
 	/// \[OPTIONAL\] The attestation key update action.
-	pub attestation_key_update: DidVerificationKeyUpdateAction,
+	pub attestation_key_update: DidFragmentUpdateAction<DidVerificationKey>,
 	/// \[OPTIONAL\] The delegation key update action.
-	pub delegation_key_update: DidVerificationKeyUpdateAction,
+	pub delegation_key_update: DidFragmentUpdateAction<DidVerificationKey>,
 	/// The set of old attestation keys to remove, given their identifiers.
 	/// If the operation also replaces the current attestation key, it will
 	/// not be considered for removal in this operation, so it is not
 	/// possible to specify it for removal in this set.
 	pub public_keys_to_remove: BTreeSet<KeyIdOf<T>>,
-	/// \[OPTIONAL\] The new endpoint URL.
-	pub new_endpoint_url: Option<Url>,
+	/// The update action on the service endpoints information.
+	pub service_endpoints_update: DidFragmentUpdateAction<ServiceEndpoints>,
 }
 
-/// Possible actions on a DID verification key within a
-/// [DidUpdateOperation].
-#[derive(Clone, Decode, Debug, Encode, Eq, Ord, PartialEq, PartialOrd)]
-pub enum DidVerificationKeyUpdateAction {
-	/// Do not change the verification key.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq)]
+pub struct ServiceEndpoints {
```

Comment:
Isn't it a single service endpoint that just has multiple URLs? ServiceEndpoints confused me a bit. I thought this are all the endpoints for a DID.

Reply:
It is a reference to a document (somewhere) containing the description of the DID service endpoints (which in this case can be replicated in up to 3 different locations).